src/control/positionControl/PositionControl.cpp

```diff
@@ -6,44 +6,215 @@
+        std::pair<RobotCommand, std::optional<Vector2>> robotCommandCollisionPair;
+        // Currently calculate all paths again on each tick because the way the path is used in control is not made for the BBT
+        if (true || (!computedPathsBB.contains(robotId) ||
+            (targetPosition - computedPathsBB[robotId].getPosition(computedPathsBB[robotId].getTotalTime())).length() > stp::control_constants::GO_TO_POS_ERROR_MARGIN ||
+            worldObjects.getFirstCollision(computedPathsBB[robotId], robotId).has_value())) {
+
+            //Create path to original target
+            computedPathsBB[robotId] = BB::BBTrajectory2D(currentPosition, currentVelocity, targetPosition,
+                                                          ai::Constants::MAX_VEL(), ai::Constants::MAX_ACC_UPPER());
+
+            //Check path to original target for collisions
+            firstCollision = worldObjects.getFirstCollision(computedPathsBB[robotId], robotId);
+
+            if (firstCollision.has_value()) {
+                robotCommandCollisionPair.second = firstCollision->collisionPosition;
+                //Create intermediate points, return a collision-free path originating from the best option of these points
+                auto newPath = findNewPath(field, robotId, currentPosition, currentVelocity, firstCollision, targetPosition,
+                                           timeStep);
+                if (newPath.has_value()) {
+                    computedPathsBB[robotId] = newPath.value();
+                }
+            }
+            computedPaths[robotId] = computedPathsBB[robotId].getPathApproach(0.2);
+        }
+        interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::yellow, robotId, interface::Drawing::LINES_CONNECTED);
+        interface::Input::drawData(interface::Visual::PATHFINDING, {computedPaths[robotId].front(), currentPosition}, Qt::darkMagenta, robotId, interface::Drawing::LINES_CONNECTED);
+        interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::magenta, robotId, interface::Drawing::DOTS);
+
+        // TODO: let the robot properly follow the BBT
+        //Current method is very hacky
+        // If you are closer to the target than the first point of the approximated path, remove it
+        if((targetPosition - currentPosition).length() < (targetPosition-computedPaths[robotId].front()).length()){
+            computedPaths[robotId].erase(computedPaths[robotId].begin());
+        }
+
+        robotCommandCollisionPair.first = RobotCommand();
+        robotCommandCollisionPair.first.pos = computedPaths[robotId].front();
+        Position trackingVelocity = pathTrackingAlgorithm.trackPathDefaultAngle(currentPosition, currentVelocity,computedPaths[robotId], robotId,pidType);
+        robotCommandCollisionPair.first.vel = Vector2(trackingVelocity.x, trackingVelocity.y);
+        robotCommandCollisionPair.first.angle = trackingVelocity.rot;
```

Comment:
This seems like you need to define a type instead of using a std::pair. You should almost never be doing this, as you have std::make_pair for making pairs.

Reply:
I removed the robotCommandCollisionPair and replaced it with a struct in WorldObjects